libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton/AttributeFilterButtonUtils.ts

```diff
@@ -0,0 +1,217 @@
+// (C) 2022 GoodData Corporation
+import { IAnalyticalBackend, IAttributeElement } from "@gooddata/sdk-backend-spi";
+import {
+    attributeElementsToAttributeElementArray,
+    getAllExceptTitle,
+    getAllTitle,
+    getFilteringTitleIntl,
+    getItemsTitles,
+    getLoadingTitleIntl,
+    getNoneTitleIntl,
+    getObjRef,
+    getValidElementsFilters,
+    isParentFilteringEnabled,
+    isParentFiltersElementsByRef,
+} from "../utils/AttributeFilterUtils";
+import {
+    filterAttributeElements,
+    IAttributeFilter,
+    isAttributeElementsByRef,
+    isNegativeAttributeFilter,
+    newNegativeAttributeFilter,
+    newPositiveAttributeFilter,
+    ObjRef,
+} from "@gooddata/sdk-model";
+import { UseCancelablePromiseStatus } from "@gooddata/sdk-ui";
+import isEmpty from "lodash/isEmpty";
+import { IntlShape } from "react-intl";
+import isNil from "lodash/isNil";
+
+export const getInitialSelectedOptions = (currentFilter: IAttributeFilter): IAttributeElement[] => {
+    // the as any cast is ok here, the data will get fixed once the element load completes
+    // this serves only to have some initial state here so that when full element data is loaded
+    // it automatically sets the props.filter.elements as selected
+    return currentFilter
+        ? (attributeElementsToAttributeElementArray(filterAttributeElements(currentFilter)) as any)
+        : [];
+};
+
+export const getInitialIsInverted = (currentFilter: IAttributeFilter): boolean => {
+    return currentFilter ? isNegativeAttributeFilter(currentFilter) : true;
+};
+
+export const isCancelablePromisePending = (cancelablePromiseStatus: UseCancelablePromiseStatus) => {
+    return cancelablePromiseStatus === "pending";
+};
+
+export const isCancelablePromiseLoading = (cancelablePromiseStatus: UseCancelablePromiseStatus) => {
+    return cancelablePromiseStatus === "loading";
+};
+
+export const isCancelablePromisePendingOrLoading = (cancelablePromiseStatus: UseCancelablePromiseStatus) => {
+    return (
+        isCancelablePromiseLoading(cancelablePromiseStatus) ||
+        isCancelablePromisePending(cancelablePromiseStatus)
+    );
+};
+
+export const getBackend = (backend: IAnalyticalBackend, props: any) => {
+    return backend.withTelemetry("AttributeFilter", props);
+};
+
+export const prepareElementsQuery = (
```

Comment:
Should we put this as a local function to `useLoadMissingData.ts`? (it's not used anywhere else)